applications/StructuralMechanicsApplication/custom_conditions/line_load_condition_2d.cpp

```diff
@@ -103,217 +103,195 @@ void LineLoadCondition2D::CalculateAll(
-
-    const Matrix& Ncontainer = GetGeometry().ShapeFunctionsValues(integration_method);
+    const IntegrationMethod integration_method = IntegrationUtilities::GetIntegrationMethodForExactMassMatrixEvaluation(r_geometry);
+    const GeometryType::IntegrationPointsArrayType& integration_points = r_geometry.IntegrationPoints(integration_method);
+    const GeometryType::ShapeFunctionsGradientsType& rDN_De = r_geometry.ShapeFunctionsLocalGradients(integration_method);
+    const Matrix& rNcontainer = r_geometry.ShapeFunctionsValues(integration_method);
 
     // Sizing work matrices
     Vector pressure_on_nodes = ZeroVector( number_of_nodes );
 
     // Pressure applied to the element itself
     double pressure_on_condition = 0.0;
-    if( this->Has( PRESSURE ) )
-    {
+    if( this->Has( PRESSURE ) ) {
         pressure_on_condition += this->GetValue( PRESSURE );
     }
-    if( this->Has( NEGATIVE_FACE_PRESSURE ) )
-    {
+    if( this->Has( NEGATIVE_FACE_PRESSURE ) ) {
         pressure_on_condition += this->GetValue( NEGATIVE_FACE_PRESSURE );
     }
-    if( this->Has( POSITIVE_FACE_PRESSURE ) )
-    {
+    if( this->Has( POSITIVE_FACE_PRESSURE ) ) {
         pressure_on_condition -= this->GetValue( POSITIVE_FACE_PRESSURE );
     }
 
-    for ( unsigned int i = 0; i < pressure_on_nodes.size(); i++ )
-    {
+    for ( IndexType i = 0; i < pressure_on_nodes.size(); i++ ) {
         pressure_on_nodes[i] = pressure_on_condition;
-        if( GetGeometry()[i].SolutionStepsDataHas( NEGATIVE_FACE_PRESSURE) )
-        {
-            pressure_on_nodes[i] += GetGeometry()[i].FastGetSolutionStepValue( NEGATIVE_FACE_PRESSURE );
+        if( r_geometry[i].SolutionStepsDataHas( NEGATIVE_FACE_PRESSURE) ) {
+            pressure_on_nodes[i] += r_geometry[i].FastGetSolutionStepValue( NEGATIVE_FACE_PRESSURE );
         }
-        if( GetGeometry()[i].SolutionStepsDataHas( POSITIVE_FACE_PRESSURE) )
-        {
-            pressure_on_nodes[i] -= GetGeometry()[i].FastGetSolutionStepValue( POSITIVE_FACE_PRESSURE );
+        if( r_geometry[i].SolutionStepsDataHas( POSITIVE_FACE_PRESSURE) ) {
+            pressure_on_nodes[i] -= r_geometry[i].FastGetSolutionStepValue( POSITIVE_FACE_PRESSURE );
         }
     }
 
     // Vector with a loading applied to the elemnt
     array_1d<double, 3 > line_load = ZeroVector(3);
-    if( this->Has( LINE_LOAD ) )
-    {
+    if( this->Has( LINE_LOAD ) ) {
         noalias(line_load) = this->GetValue( LINE_LOAD );
     }
 
-    for ( unsigned int point_number = 0; point_number < integration_points.size(); point_number++ )
-    {
-        const double det_j = GetGeometry().DeterminantOfJacobian( integration_points[point_number] );
-
+    // Declaring tangent and Jacobian
+    array_1d<double, 3> tangent_xi, tangent_eta;
+    tangent_eta[0] = 0.0;
+    tangent_eta[1] = 0.0;
+    tangent_eta[2] = 1.0;
+    Matrix J(2, 1);
```

Comment:
We use the Jacobian in order to be able to consider geometries different than Line2D2